docs/data/sql_functions.yml

```diff
@@ -665,6 +665,9 @@ json:
       // STRING: "false"
       JSON_VALUE('{"a": true}', 'strict $.b'
           DEFAULT FALSE ON ERROR)
+
+      // DOUBLE: 0.998
+      JSON_VALUE('{"a": {"b.c": [0.998,0.996]}}','$.a.[''b.c''][0]' RETURNING DOUBLE)
```

Comment:
Since FLINK-24271 is about adding an example on how to access keys with "non-trivial" names, how about keeping the example as simple as possible to focus on actually showing how to do that? For example, like this:

```
// INT: 1
JSON_VALUE('{"a.b": 1}', '$.[''a.b'']' RETURNING INT)
```

Please also run the docs locally and make sure that the line length of the example doesn't cause the code block to scroll. If it does, line-break the example (as I did in the examples above) to prevent it from scrolling.

Reply:
For what it's worth, there is nothing special about this in terms of JSON functions. This is just a case of escaping a single quote within a SQL string.